kolibri/plugins/user/assets/src/state/actions.js

```diff
@@ -75,6 +75,27 @@ export function updateUserProfile(store, { edits, session }) {
   );
 }
 
+export function updateUserProfilePassword(store, password) {
+  const session = store.state.core.session;
+  const savedUserModel = FacilityUserResource.getModel(session.user_id);
+
+  store.dispatch('SET_PROFILE_BUSY', true);
+
+  return savedUserModel.save({ password }).then(
+    () => {
+      store.dispatch('SET_PROFILE_BUSY', false);
+      store.dispatch('SET_PROFILE_PASSWORD_MODAL', false);
+      store.dispatch('SET_PROFILE_PASSWORD_SUCCESS', true);
+      // snackbars are typically used with core state properties. Decided not to do that here.
+      window.setTimeout(() => store.dispatch('SET_PROFILE_PASSWORD_SUCCESS', false), 4000);
```

Comment:
@indirectlylit flagging this. Snackbar wouldn't disappear. Upon investigation, found out that duration dismissal is not really meant to be used with `v-if`s. Because this is used in core for the disconnection things, it uses some core state properties. Thought that'd be overkill.

Reply:
My recommendation would be to piggy-pack on top of the existing global snackbars. Maybe something like...

* Rename `<connection-snackbars />` to something like `<global-snackbars />`. Add a new `core-snackbar` in there which can render arbitrary text.
* Add some new global state like  `state.core.transientNotificationText`.
* Make a new global action like `setTransientNotificationText`. Under the hood, this:
  * Updates  `state.core.transientNotificationText` and sets `state.core.currentSnackbar` to `ConnectionSnackbars.TRANSIENT_NOTIFICATION`
  * kicks off a timer which, when complete, clears the notifcation

That way, we can set notifications from everywhere rather than re-implementing it for every use-case.

thoughts?

(cc @christianmemije)

Reply:
The autodismiss did not work, because it sets `core.currentSnackbar` to null after the timeout. So here you can do store.dispatch('CORE_SET_CURRENT_SNACKBAR', PASSWORD_CHANGE_SUCCESS); and then in your v-if check if core.currenSnackbar ===  PASSWORD_CHANGE_SUCCESS.

Reply:
> re-implementing it for every use-case.

What are you referring to when you say `re-implementing`? As in adding a core-snackbar component?

Reply:
> What are you referring to when you say re-implementing?

Below, [we added](https://github.com/learningequality/kolibri/pull/3097/files#diff-3705966c69fdd78812ff7c734e1b92c7R103) a new `core-snackbar` just to handle this one specific case of "Password successfully changed".

I thought the idea though was to eventually show all kinds of transient notifications resulting from successful actions in modals? If so, we shouldn't keep adding more component instances and state like `state.pageState.passwordState.success` for every case.

Reply:
So this would be for plain text notifications that do not have actions associated with them such as undo? :+1:

Reply:
Yeah, that was what I was thinking

Reply:
Cool. I like the idea of having a `<global-snackbars />` component. However, instead of combining it with the `<connection-snackbars />`, include the `<connecton-snackbars />` since that component has a lot of connection specific logic. And no need to add an extra timer, :autoDismiss="true" could take care of that.

Reply:
@DXCanas You can now make use of the new snackbar implementation introduced in #3149